lib/util.ml

```diff
@@ -69,16 +70,41 @@ module Option = struct
     | None -> default
 end
 
-module List = struct
-  let find_map f l =
-    let rec aux = function
-      | [] -> None
-      | h :: t ->
-        match f h with
-        | Some x -> Some x
-        | None -> aux t
-    in
-    aux l
+module Sexp = struct
+  type t =
+    | Atom of string
+    | List of t list
+
+  let rec equal t t' =
```

Comment:
Isn't this function behaving exactly the same as `(=)` ?

Otherwise your changes look good to me.

Reply:
It should behave the same but I think it's best practice to redefine them. `Stdlib.(=)` is an external C function. I can't remember why it's best to define your own though, @Julow might know more about that!

Reply:
`=` is a bit slower indeed. The main reason to avoid it is when there is polymorphism (user should be able to give its equal function), or when there is values that cannot be compared (eg. functions).
In this case, `=` is fine.

Reply:
I stole this one from `stdune` to be honest. Should I remove it then or is it fine to keep it?

Reply:
I think it's fine to keep it too.
It would be better to import the whole module from stdune to ensures that it's working correctly but that would bring a handful of other modules with it.
I'm Ok with the current code.